src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+          .put(401, "Request not authorized.")
+          .put(403, "Client does not have sufficient privileges.")
+          .put(404, "Failed to find the resource.")
+          .put(409, "The resource already exists.")
+          .put(429, "Request throttled by the backend server.")
+          .put(500, "Internal server error.")
+          .put(503, "Backend servers are over capacity. Try again later.")
+          .build();
+
+  private final FirebaseApp app;
+  private final JsonFactory jsonFactory;
+  private final HttpRequestFactory requestFactory;
+  private final ScheduledExecutorService scheduledExecutor;
+
+  private final CreateAndroidAppFromAppIdFunction createAndroidAppFromAppIdFunction =
+      new CreateAndroidAppFromAppIdFunction();
+  private final CreateIosAppFromAppIdFunction createIosAppFromAppIdFunction =
+      new CreateIosAppFromAppIdFunction();
+
+  private HttpResponseInterceptor interceptor;
+
+  FirebaseProjectManagementServiceImpl(FirebaseApp app) {
+    this.app = app;
+
+    JsonFactory jsonFactory = app.getOptions().getJsonFactory();
+    this.jsonFactory = jsonFactory;
+
+    HttpTransport httpTransport = app.getOptions().getHttpTransport();
+    this.requestFactory =
+        httpTransport.createRequestFactory(new FirebaseRequestInitializer(app));
+
+    this.scheduledExecutor = new FirebaseScheduledExecutor(
+        ImplFirebaseTrampolines.getThreadFactory(app), "firebase-project-management");
+  }
+
+  @VisibleForTesting
+  void setInterceptor(HttpResponseInterceptor interceptor) {
+    this.interceptor = interceptor;
+  }
+
+  void destroy() {
+    // NOTE: Any instance of IosApp, AndroidApp, or FirebaseProjectManagement that depends on this
+    // instance will no longer be able to make RPC calls. All polling or waiting iOS or Android App
+    // creations will be interrupted, even though the initial creation RPC (if made successfully) is
+    // still processed normally (asynchronously) by the server.
+    scheduledExecutor.shutdownNow();
+  }
+
+  /* getAndroidApp */
+
+  @Override
+  public AndroidAppMetadata getAndroidApp(String appId) throws FirebaseProjectManagementException {
+    return getAndroidAppOp(appId).call();
+  }
+
+  @Override
+  public ApiFuture<AndroidAppMetadata> getAndroidAppAsync(String appId) {
+    return getAndroidAppOp(appId).callAsync(app);
+  }
+
+  private CallableOperation<AndroidAppMetadata, FirebaseProjectManagementException> getAndroidAppOp(
+      final String appId) {
+    return new CallableOperation<AndroidAppMetadata, FirebaseProjectManagementException>() {
+      @Override
+      protected AndroidAppMetadata execute() throws FirebaseProjectManagementException {
+        String url = String.format(
+            "%s/v1beta1/projects/-/androidApps/%s", FIREBASE_PROJECT_MANAGEMENT_URL, appId);
+        AndroidAppResponse parsedResponse = new AndroidAppResponse();
+        makeGetRequest(url, parsedResponse, appId, "App ID");
+        return new AndroidAppMetadata(
```

Comment:
I think AndroidAppMetadata should just provide a constructor that accepts an AndroidAppResponse as the only argument.